lightning/src/onion_message/messenger.rs

```diff
@@ -563,6 +565,10 @@ where
 		// recipient's node_id.
 		const MIN_PEER_CHANNELS: usize = 3;
 
+		// Add a random number (1 to 5) of dummy hops to each blinded path
+		// to make it harder to infer the recipient's position.
+		let dummy_hops_count = rand::thread_rng().gen_range(1..5);
```

Comment:
Should use `entropy_source` instead of introducing the `rand` dep. Also, if `compact_paths` is set we shouldn't set any dummy hops.

Reply:
Thanks for the pointer! Updated the code to remove the `rand` dependency, use entropy_source, and not use dummy hops for compact paths in [pr3728.03](https://github.com/shaavan/rust-lightning/commits/pr3728.03)